src/config/cli_options.rs

```diff
@@ -0,0 +1,144 @@
+use getopts::{Matches, Options};
```

Comment:
have you also looked at `clap` ?
https://docs.rs/clap/latest/clap/

Reply:
Initially I looked at clap and found a bit cryptic and bloated, but the code is actually more succinct now. The resulting binary is now 200k bigger tho. Still an okay trade-off for now.